src/partisan_hyparview_peer_service_manager.erl

```diff
@@ -1248,20 +1248,39 @@ prwl() ->
     partisan_config:get(prwl, 6).
 
 %% @private
+-spec remove_from_passive_view(node_spec(), passive()) -> passive().
 remove_from_passive_view(Peer, Passive) ->
     sets:del_element(Peer, Passive).
 
 %% @private
+-spec is_in_passive_view(node_spec() | node(), passive()) ->
+        false | {true, node_spec()}.
 is_in_passive_view(Peer, Passive) ->
-    sets:is_element(Peer, Passive).
+    is_in_view(Peer, Passive).
 
 %% @private
+-spec remove_from_active_view(node_spec(), active()) -> active().
 remove_from_active_view(Peer, Active) ->
     sets:del_element(Peer, Active).
 
 %% @private
+-spec is_in_active_view(node_spec() | node(), active()) ->
+        false | {true, node_spec()}.
 is_in_active_view(Peer, Active) ->
-    sets:is_element(Peer, Active).
+    is_in_view(Peer, Active).
+
+-spec is_in_view(node_spec() | node(), active() | passive()) ->
+        false | {true, node_spec()}.
+is_in_view({_Node, _, _} = Peer, View) ->
+    case sets:is_element(Peer, View) of
+        false -> false;
+        true -> {true, Peer}
+    end;
+is_in_view(Node, View) ->
```

Comment:
Which of these views is not using the three tuple format?  That seems like a bug.